server/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java

```diff
@@ -207,20 +225,15 @@ protected Directory newDirectory(Path dir) throws IOException {
         return new SimpleFSDirectory(dir);
     }
 
-    private void cleanupOldFiles(final String prefix, final String currentStateFile, Path[] locations) throws IOException {
-        final DirectoryStream.Filter<Path> filter = entry -> {
-            final String entryFileName = entry.getFileName().toString();
-            return Files.isRegularFile(entry)
-                    && entryFileName.startsWith(prefix) // only state files
-                    && currentStateFile.equals(entryFileName) == false; // keep the current state file around
-        };
-        // now clean up the old files
-        for (Path dataLocation : locations) {
-            logger.trace("cleanupOldFiles: cleaning up {}", dataLocation);
-            try (DirectoryStream<Path> stream = Files.newDirectoryStream(dataLocation.resolve(STATE_DIR_NAME), filter)) {
-                for (Path stateFile : stream) {
-                    Files.deleteIfExists(stateFile);
-                    logger.trace("cleanupOldFiles: cleaned up {}", stateFile);
+    private void cleanupOldFiles(final String currentStateFile, Path[] locations) throws IOException {
+        for (Path location: locations){
+            Path stateLocation = location.resolve(STATE_DIR_NAME);
+            try (Directory stateDir = newDirectory(stateLocation)){
+                for (String file : stateDir.listAll()){
```

Comment:
Tip, just in case: you can let intellij auto format a selection (and not the entire file) but selecting it and choosing reformat code from the Code menu (or using whatever keyboard shortcut you have).

Reply:
Sorry about that, I know about format option and use it, but sometimes forget to apply it. Unfortunately, there is no good way for IDEA to do it safe and automatically for you